compiler/dyno/CMakeLists.txt

```diff
@@ -30,6 +30,7 @@ option(DYNO_LLVM_LINK_ARGS "Override linker args for LLVM")
 # according to policy CMP0004
 string(STRIP ${DYNO_LLVM_COMP_ARGS} DYNO_LLVM_COMP_ARGS)
 string(STRIP ${DYNO_LLVM_LINK_ARGS} DYNO_LLVM_LINK_ARGS)
+string(REPLACE "-ljemalloc" "-ljemalloc_pic" DYNO_LLVM_LINK_ARGS ${DYNO_LLVM_LINK_ARGS})
```

Comment:
Naive question, but when do you need the `${}`? Seems like the order is reversed between `REPLACE` and `STRIP`, but I don't know the calling conventions

Reply:
Also, do you know why dyno requires `PIC`, but production compiler doesn't"

Reply:
@arezaii - it might be a stretch, but maybe this is related to something you were facing with cmake'ing `chpl` ?

Certainly, the dyno library is a `.so` and that requires position independent code in order to build. However, it's not obvious to me why that would need to impact `chpl` or `chpldoc` themselves. But, I'm frequently confused about these requirements.

Could be related to this code:

https://github.com/chapel-lang/chapel/blob/4848643f775e52f0d4b7779bc8136d2f6ccaa8f1/compiler/dyno/lib/CMakeLists.txt#L21-L23

@DanilaFe - is your modification here adjusting the `.so` to include jemalloc statically? I would think that you would not want to do that & instead whatever links with `liblibdyno.so` could/should also link with LLVM and jemalloc etc. Are we building a `jemalloc.so` for host configurations? If so, that might allow us to make this distinction moot.

Reply:
From https://github.com/Cray/chapel-private/issues/2558#issuecomment-952140130, I think it's just a matter of getting rpath to work. Either way @DanilaFe - I think you should either get jemalloc linking dynamically here or move the `-l` flags to the `chpldoc` (and `chpl`, after @arezaii's work) link step. (Or I am misunderstanding something).

Reply:
I've been trying to switch to only using `-jemalloc` for the tests and `chpldoc`, but as far as I can tell nothing is happening. `jemalloc` is supposed to print something when you link against it and specify `MALLOC_CONF=stats_print:true`, but it prints nothing. I have no idea how to test if it's there; the only evidence I see is the flag in `make VERBOSE=1` output:
```
/usr/bin/g++   -g   -Wall -Werror -Wpointer-arith -Wwrite-strings -Wno-strict-aliasing -Wno-comment -Wmissing-braces -Wsuggest-override -Wno-error=suggest-override -Wmissing-declarations -I. -I../include/cray-xc -I../include -I../../build/compiler/cray-xc/gnu/x86_64/hostmem-jemalloc/llvm-bundled/14/san-none -I../../runtime/include/encoding -I/lus/cls01029/fedorin/chapel/third-party/utf8-decoder -I../dyno/include   -I/lus/cls01029/fedorin/chapel/third-party/llvm/install/cray-xc-x86_64/include -std=c++14 -fno-exceptions -fno-rtti -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Wno-comment -DHAVE_LLVM -I/lus/scratch/fedorin/chapel/third-party/jemalloc/install/host/cray-xc-x86_64-gnu/include    -Wall -Werror -Wpointer-arith -Wwrite-strings -Wno-strict-aliasing -Wno-comment -Wmissing-braces -Wsuggest-override -Wno-error=suggest-override -Wmissing-declarations -g   -static-libstdc++ CMakeFiles/testParseReturn.dir/testParseReturn.cpp.o ../CMakeFiles/tests-common.dir/common.cpp.o  -o testParseReturn  -Wl,-rpath,/lus/scratch/fedorin/chapel/build/compiler/cray-xc/gnu/x86_64/hostmem-jemalloc/llvm-bundled/14/san-none/dyno/Debug/lib -ljemalloc ../../lib/liblibdyno.so -L/lus/cls01029/fedorin/chapel/third-party/llvm/install/cray-xc-x86_64/lib -Wl,-rpath,/lus/cls01029/fedorin/chapel/third-party/llvm/install/cray-xc-x86_64/lib -lclangFrontend -lclangSerialization -lclangDriver -lclangCodeGen -lclangParse -lclangSema -lclangAnalysis -lclangEdit -lclangASTMatchers -lclangAST -lclangLex -lclangBasic -L/lus/cls01029/fedorin/chapel/third-party/llvm/install/cray-xc-x86_64/lib -lLLVMLTO -lLLVMPasses -lLLVMExtensions -lLLVMCoroutines -lLLVMCoverage -lLLVMNVPTXCodeGen -lLLVMNVPTXDesc -lLLVMNVPTXInfo -lLLVMAArch64Disassembler -lLLVMAArch64AsmParser -lLLVMAArch64CodeGen -lLLVMAArch64Desc -lLLVMAArch64Utils -lLLVMAArch64Info -lLLVMX86TargetMCA -lLLVMMCA -lLLVMX86Disassembler -lLLVMX86AsmParser -lLLVMX86CodeGen -lLLVMCFGuard -lLLVMGlobalISel -lLLVMX86Desc -lLLVMX86Info -lLLVMMCDisassembler -lLLVMSelectionDAG -lLLVMAsmPrinter -lLLVMDebugInfoMSF -lLLVMCodeGen -lLLVMTarget -lLLVMObjCARCOpts -lLLVMOption -lLLVMipo -lLLVMInstrumentation -lLLVMVectorize -lLLVMLinker -lLLVMIRReader -lLLVMAsmParser -lLLVMFrontendOpenMP -lLLVMScalarOpts -lLLVMInstCombine -lLLVMAggressiveInstCombine -lLLVMTransformUtils -lLLVMBitWriter -lLLVMAnalysis -lLLVMProfileData -lLLVMDebugInfoDWARF -lLLVMObject -lLLVMTextAPI -lLLVMMCParser -lLLVMMC -lLLVMDebugInfoCodeView -lLLVMBitReader -lLLVMCore -lLLVMRemarks -lLLVMBinaryFormat -lLLVMBitstreamReader -lLLVMSupport -lLLVMDemangle -L/lus/scratch/fedorin/chapel/third-party/jemalloc/install/host/cray-xc-x86_64-gnu/lib  -lrt -ldl -lz -ltinfo -lxml2 -lm -lpthread
```
How do I verify anything I'm doing has any effect?

Reply:
I typically do something like `MALLOC_CONF="abort_conf:true,some_invalid_option"` to verify that jemalloc is being used. For the target allocator we prefix with `CHPL_JE_` so it'd be `CHPL_JE_MALLOC_CONF=..` but we don't prefix for the host allocator since we're trying to override the std allocators.

Reply:
@ronawho: confirmed that it doesn't work, then 😄. `-ljemalloc` is specified in the linker options, so I've no idea why it's not working.

@mppf: In that case I will just strip `-ljemalloc` from the compile flags.